src/SOS/Strike/dbgengservices.cpp

```diff
@@ -189,10 +189,31 @@ DbgEngServices::GetDebuggeeType(
 }
 
 HRESULT
-DbgEngServices::GetExecutingProcessorType(
+DbgEngServices::GetProcessorType(
     PULONG type)
 {
-    return m_control->GetExecutingProcessorType(type);
+    ULONG executingType;
+    HRESULT hr = m_control->GetExecutingProcessorType(&executingType);
+    if (FAILED(hr))
+    {
+        return hr;
+    }
+    _ASSERTE(executingType != IMAGE_FILE_MACHINE_ARM64X);
+    *type = executingType;
+#if defined(SOS_TARGET_AMD64) || defined(SOS_TARGET_ARM64)
+    if (executingType == IMAGE_FILE_MACHINE_ARM64EC)
+    {
+        ULONG actualType;
+        if (SUCCEEDED(m_control->GetActualProcessorType(&actualType)))
+        {
+            if (actualType == IMAGE_FILE_MACHINE_AMD64 || actualType == IMAGE_FILE_MACHINE_ARM64)
```

Comment:
This doesn't work with what we saw earlier.

Reply:
Should I remove the GetActualProcessorType() and always return AMD64 for the ARM64EC case.

Reply:
I removed the GetActualProcessorType() calls.